source/extensions/filters/http/cache/cache_headers_utils.cc

```diff
@@ -0,0 +1,182 @@
+      max_stale_(max_stale) {}
+
+RequestCacheControl::RequestCacheControl(absl::string_view cache_control_header) {
+  must_validate_ = no_store_ = no_transform_ = only_if_cached_ = false;
+  std::vector<absl::string_view> directives = absl::StrSplit(cache_control_header, ',');
+
+  for (auto full_directive : directives) {
+    absl::string_view directive, argument;
+    std::tie(directive, argument) = separateDirectiveAndArgument(full_directive);
+
+    if (directive == "no-cache") {
+      must_validate_ = true;
+    } else if (directive == "no-store") {
+      no_store_ = true;
+    } else if (directive == "no-transform") {
+      no_transform_ = true;
+    } else if (directive == "only-if-cached") {
+      only_if_cached_ = true;
+    } else if (directive == "max-age") {
+      max_age_ = parseDuration(argument);
+    } else if (directive == "min-fresh") {
+      min_fresh_ = parseDuration(argument);
+    } else if (directive == "max-stale") {
+      max_stale_ = argument.empty() ? SystemTime::duration::max() : parseDuration(argument);
+    }
+  }
+}
+
+ResponseCacheControl::ResponseCacheControl(bool must_validate, bool no_store, bool no_transform,
+                                           bool no_stale, bool is_public, OptionalDuration max_age)
+    : must_validate_(must_validate), no_store_(no_store), no_transform_(no_transform),
+      no_stale_(no_stale), is_public_(is_public), max_age_(max_age) {}
+
+ResponseCacheControl::ResponseCacheControl(absl::string_view cache_control_header) {
+  must_validate_ = no_store_ = no_transform_ = no_stale_ = is_public_ = false;
+  std::vector<absl::string_view> directives = absl::StrSplit(cache_control_header, ',');
+
+  for (auto full_directive : directives) {
+    absl::string_view directive, argument;
+    std::tie(directive, argument) = separateDirectiveAndArgument(full_directive);
+
+    if (directive == "no-cache") {
+      // If no-cache directive has arguments they are ignored - not handled
+      must_validate_ = true;
+    } else if (directive == "must-revalidate" || directive == "proxy-revalidate") {
+      no_stale_ = true;
+    } else if (directive == "no-store" || directive == "private") {
+      // If private directive has arguments they are ignored - not handled
+      no_store_ = true;
+    } else if (directive == "no-transform") {
+      no_transform_ = true;
+    } else if (directive == "public") {
+      is_public_ = true;
+    } else if (directive == "s-maxage") {
+      max_age_ = parseDuration(argument);
+    } else if (!max_age_.has_value() && directive == "max-age") {
+      max_age_ = parseDuration(argument);
+    }
+  }
+}
+
+std::ostream& operator<<(std::ostream& os, const OptionalDuration& duration) {
+  return duration.has_value() ? os << duration.value().count() : os << " ";
+}
+
+std::ostream& operator<<(std::ostream& os, const RequestCacheControl& request_cache_control) {
+  return os << "{"
```

Comment:
Keep them this way if you like, but please consider instead only reporting the "true" bits (e.g. "no-store, min_fresh=12")

Reply:
After the cache validation PR I will dedicate a PR to cleaning and improving tests and coverage. One of my todos for this is to move these operators to the test files (as they are only used for testing) and clean them. I would rather postpone this for that PR.

Reply:
Also the presence of these operators in the source code makes the coverage appear to be lower than it actually is, as these operators are only invoked on test failures.

Reply:
SGTM